### PR TITLE
Rename/update API exotica_python

### LINKED ISSUE #460: 'getSolver' does not return a solver
The method `Scene::getSolver()` is rather badly named since it actually returns a `KinematicTree` and not a solver as in `exotica::MotionSolver`.

This is especially confusing if used with the Python bindings, since after setting up a solver and a problem like:
```Python
self.solver = exo.Setup.createSolver(...)
self.problem = exo.Setup.createProblem(...)
self.solver.specifyProblem(self.problem)
```
a further call to:
```Python
problem_solver = self.problem.getScene().getSolver()
```
suggests that `problem_solver` is the same instance as the `self.solver` defined above.

The method `Scene::getSolver()` should be renamed to something like `Scene::getKinematicTree()`.

### PR DESCRIPTION
**Pull request by wxmerkt**
- Renames entire API in `exotica_python` (minor exception where it mirrors KDL C++ API)
- Updates all examples in `exotica_examples` to comply
- Fixes some examples that were broken

Relates to #421 

Fixes #460

### CODE REVIEW

File: exotica_python/src/Exotica.cpp [hunk 1]
```diff
@@ -631,468 +599,450 @@ PYBIND11_MODULE(_pyexotica, module)
...
+    unconstrained_time_indexed_problem.def_readonly("JN", &UnconstrainedTimeIndexedProblem::JN);
+    unconstrained_time_indexed_problem.def_readonly("N", &UnconstrainedTimeIndexedProblem::N);
+    unconstrained_time_indexed_problem.def_readonly("num_tasks", &UnconstrainedTimeIndexedProblem::NumTasks);
+    unconstrained_time_indexed_problem.def_readonly("Phi", &UnconstrainedTimeIndexedProblem::Phi);
+    unconstrained_time_indexed_problem.def_readonly("J", &UnconstrainedTimeIndexedProblem::J);
+    unconstrained_time_indexed_problem.def("get_scalar_task_cost", &UnconstrainedTimeIndexedProblem::getScalarTaskCost);
+    unconstrained_time_indexed_problem.def("get_scalar_task_jacobian", &UnconstrainedTimeIndexedProblem::getScalarTaskJacobian);
+    unconstrained_time_indexed_problem.def("get_scalar_transition_cost", &UnconstrainedTimeIndexedProblem::getScalarTransitionCost);
+    unconstrained_time_indexed_problem.def("get_scalar_transition_jacobian", &UnconstrainedTimeIndexedProblem::getScalarTransitionJacobian);
+    unconstrained_time_indexed_problem.def_readonly("Cost", &UnconstrainedTimeIndexedProblem::Cost);
```
Comment (VladimirIvan) [inline exotica_python/src/Exotica.cpp]: 'cost' - not a math symbol

File: exotica_python/src/Exotica.cpp [hunk 2]
```diff
@@ -631,468 +599,450 @@ PYBIND11_MODULE(_pyexotica, module)
...
+    py::class_<SamplingTask, std::shared_ptr<SamplingTask>> sampling_task(module, "SamplingTask");
+    sampling_task.def_readonly("PhiN", &SamplingTask::PhiN);
+    sampling_task.def_readonly("JN", &SamplingTask::JN);
+    sampling_task.def_readonly("num_tasks", &SamplingTask::NumTasks);
+    sampling_task.def_readonly("y", &SamplingTask::y);
+    sampling_task.def_readonly("ydiff", &SamplingTask::ydiff);
+    sampling_task.def_readonly("Phi", &SamplingTask::Phi);
+    sampling_task.def_readonly("S", &SamplingTask::S);
+    sampling_task.def_readonly("Tasks", &SamplingTask::Tasks);
+    sampling_task.def_readonly("TaskMaps", &SamplingTask::TaskMaps);
```
Comment (VladimirIvan) [inline exotica_python/src/Exotica.cpp]: Use same API as `planning_problem`

File: exotica_python/src/Exotica.cpp [hunk 3]
```diff
@@ -631,468 +599,450 @@ PYBIND11_MODULE(_pyexotica, module)
...
+    end_pose_task.def_readonly("JN", &EndPoseTask::JN);
+    end_pose_task.def_readonly("num_tasks", &EndPoseTask::NumTasks);
+    end_pose_task.def_readonly("y", &EndPoseTask::y);
+    end_pose_task.def_readonly("ydiff", &EndPoseTask::ydiff);
+    end_pose_task.def_readonly("Phi", &EndPoseTask::Phi);
+    // end_pose_task.def_readonly("H", &EndPoseTask::H);
+    end_pose_task.def_readonly("J", &EndPoseTask::J);
+    end_pose_task.def_readonly("S", &EndPoseTask::S);
+    end_pose_task.def_readonly("Tasks", &EndPoseTask::Tasks);
+    end_pose_task.def_readonly("TaskMaps", &EndPoseTask::TaskMaps);
```
Comment (VladimirIvan) [inline exotica_python/src/Exotica.cpp]: Use same API as `planning_problem`

File: exotica_python/src/Exotica.cpp [hunk 4]
```diff
@@ -631,468 +599,450 @@ PYBIND11_MODULE(_pyexotica, module)
...
+    time_indexed_task.def_readonly("num_tasks", &TimeIndexedTask::NumTasks);
+    time_indexed_task.def_readonly("y", &TimeIndexedTask::y);
+    time_indexed_task.def_readonly("ydiff", &TimeIndexedTask::ydiff);
+    time_indexed_task.def_readonly("Phi", &TimeIndexedTask::Phi);
+    // time_indexed_task.def_readonly("H", &TimeIndexedTask::H);
+    time_indexed_task.def_readonly("J", &TimeIndexedTask::J);
+    time_indexed_task.def_readonly("S", &TimeIndexedTask::S);
+    time_indexed_task.def_readonly("T", &TimeIndexedTask::T);
+    time_indexed_task.def_readonly("Tasks", &TimeIndexedTask::Tasks);
+    time_indexed_task.def_readonly("TaskMaps", &TimeIndexedTask::TaskMaps);
```
Comment (VladimirIvan) [inline exotica_python/src/Exotica.cpp]: Use same API as `planning_problem`

File: exotica_python/src/Exotica.cpp [hunk 5]
```diff
@@ -631,468 +599,450 @@ PYBIND11_MODULE(_pyexotica, module)
...
-    motionSolver.def("getPlanningTime", &MotionSolver::getPlanningTime);
-    motionSolver.def("specifyProblem", &MotionSolver::specifyProblem, "Assign problem to the solver", py::arg("planningProblem"));
-    motionSolver.def(
-        "solve", [](std::shared_ptr<MotionSolver> sol) { return Solve(sol); },
+    py::class_<TaskMap, std::shared_ptr<TaskMap>, Object> task_map(module, "TaskMap");
+    task_map.def_readonly("id", &TaskMap::Id);
+    task_map.def_readonly("start", &TaskMap::Start);
+    task_map.def_readonly("length", &TaskMap::Length);
+    task_map.def_readonly("startJ", &TaskMap::StartJ);
+    task_map.def_readonly("lengthJ", &TaskMap::LengthJ);
```
Comment (VladimirIvan) [inline exotica_python/src/Exotica.cpp]: Wrrrrr. I'm angry at naming conventions!

File: exotica_python/src/Exotica.cpp [hunk 6]
```diff
@@ -631,468 +599,450 @@ PYBIND11_MODULE(_pyexotica, module)
...
+    time_indexed_problem.def("get_scalar_transition_cost", &TimeIndexedProblem::getScalarTransitionCost);
+    time_indexed_problem.def("get_scalar_transition_jacobian", &TimeIndexedProblem::getScalarTransitionJacobian);
+    time_indexed_problem.def("get_equality", &TimeIndexedProblem::getEquality);
+    time_indexed_problem.def("get_equality_jacobian", &TimeIndexedProblem::getEqualityJacobian);
+    time_indexed_problem.def("get_inequality", &TimeIndexedProblem::getInequality);
+    time_indexed_problem.def("get_inequality_jacobian", &TimeIndexedProblem::getInequalityJacobian);
+    time_indexed_problem.def("get_bounds", &TimeIndexedProblem::getBounds);
+    time_indexed_problem.def_readonly("Cost", &TimeIndexedProblem::Cost);
+    time_indexed_problem.def_readonly("Inequality", &TimeIndexedProblem::Inequality);
+    time_indexed_problem.def_readonly("Equality", &TimeIndexedProblem::Equality);
```
Comment (VladimirIvan) [inline exotica_python/src/Exotica.cpp]: Lower case

File: exotica_python/src/Exotica.cpp [hunk 7]
```diff
@@ -631,468 +599,450 @@ PYBIND11_MODULE(_pyexotica, module)
...
+    unconstrained_end_pose_problem.def_readonly("N", &UnconstrainedEndPoseProblem::N);
+    unconstrained_end_pose_problem.def_readonly("num_tasks", &UnconstrainedEndPoseProblem::NumTasks);
+    unconstrained_end_pose_problem.def_readonly("Phi", &UnconstrainedEndPoseProblem::Phi);
+    unconstrained_end_pose_problem.def_readonly("J", &UnconstrainedEndPoseProblem::J);
+    unconstrained_end_pose_problem.def_property_readonly("ydiff", [](UnconstrainedEndPoseProblem* prob) { return prob->Cost.ydiff; });
+    unconstrained_end_pose_problem.def_property("q_nominal", &UnconstrainedEndPoseProblem::getNominalPose, &UnconstrainedEndPoseProblem::setNominalPose);
+    unconstrained_end_pose_problem.def("get_scalar_cost", &UnconstrainedEndPoseProblem::getScalarCost);
+    unconstrained_end_pose_problem.def("get_scalar_jacobian", &UnconstrainedEndPoseProblem::getScalarJacobian);
+    unconstrained_end_pose_problem.def("get_scalar_task_cost", &UnconstrainedEndPoseProblem::getScalarTaskCost);
+    unconstrained_end_pose_problem.def_readonly("Cost", &UnconstrainedEndPoseProblem::Cost);
```
Comment (VladimirIvan) [inline exotica_python/src/Exotica.cpp]: Lower case

File: exotica_python/src/Exotica.cpp [hunk 8]
```diff
@@ -631,468 +599,450 @@ PYBIND11_MODULE(_pyexotica, module)
...
+    end_pose_problem.def("get_scalar_jacobian", &EndPoseProblem::getScalarJacobian);
+    end_pose_problem.def("get_scalar_task_cost", &EndPoseProblem::getScalarTaskCost);
+    end_pose_problem.def("get_equality", &EndPoseProblem::getEquality);
+    end_pose_problem.def("get_equality_jacobian", &EndPoseProblem::getEqualityJacobian);
+    end_pose_problem.def("get_inequality", &EndPoseProblem::getInequality);
+    end_pose_problem.def("get_inequality_jacobian", &EndPoseProblem::getInequalityJacobian);
+    end_pose_problem.def("get_bounds", &EndPoseProblem::getBounds);
+    end_pose_problem.def_readonly("Cost", &EndPoseProblem::Cost);
+    end_pose_problem.def_readonly("Inequality", &EndPoseProblem::Inequality);
+    end_pose_problem.def_readonly("Equality", &EndPoseProblem::Equality);
```
Comment (VladimirIvan) [inline exotica_python/src/Exotica.cpp]: Lower case

File: exotica_python/src/Exotica.cpp [hunk 9]
```diff
@@ -631,468 +599,450 @@ PYBIND11_MODULE(_pyexotica, module)
...
+    bounded_end_pose_problem.def_readonly("JN", &BoundedEndPoseProblem::JN);
+    bounded_end_pose_problem.def_readonly("N", &BoundedEndPoseProblem::N);
+    bounded_end_pose_problem.def_readonly("num_tasks", &BoundedEndPoseProblem::NumTasks);
+    bounded_end_pose_problem.def_readonly("Phi", &BoundedEndPoseProblem::Phi);
+    bounded_end_pose_problem.def_readonly("J", &BoundedEndPoseProblem::J);
+    bounded_end_pose_problem.def("get_scalar_cost", &BoundedEndPoseProblem::getScalarCost);
+    bounded_end_pose_problem.def("get_scalar_jacobian", &BoundedEndPoseProblem::getScalarJacobian);
+    bounded_end_pose_problem.def("get_scalar_task_cost", &BoundedEndPoseProblem::getScalarTaskCost);
+    bounded_end_pose_problem.def("get_bounds", &BoundedEndPoseProblem::getBounds);
+    bounded_end_pose_problem.def_readonly("Cost", &BoundedEndPoseProblem::Cost);
```
Comment (VladimirIvan) [inline exotica_python/src/Exotica.cpp]: Lower case

File: exotica_python/src/Exotica.cpp [hunk 10]
```diff
@@ -631,468 +599,450 @@ PYBIND11_MODULE(_pyexotica, module)
...
+    py::class_<SamplingProblem, std::shared_ptr<SamplingProblem>, PlanningProblem> sampling_problem(prob, "SamplingProblem");
+    sampling_problem.def("update", &SamplingProblem::Update);
+    sampling_problem.def_property("goal_state", &SamplingProblem::getGoalState, &SamplingProblem::setGoalState);
+    sampling_problem.def("get_space_dim", &SamplingProblem::getSpaceDim);
+    sampling_problem.def("get_bounds", &SamplingProblem::getBounds);
+    sampling_problem.def_readonly("N", &SamplingProblem::N);
+    sampling_problem.def_readonly("num_tasks", &SamplingProblem::NumTasks);
+    sampling_problem.def_readonly("Phi", &SamplingProblem::Phi);
+    sampling_problem.def_readonly("Inequality", &SamplingProblem::Inequality);
+    sampling_problem.def_readonly("Equality", &SamplingProblem::Equality);
```
Comment (VladimirIvan) [inline exotica_python/src/Exotica.cpp]: Lower case (keep Phi)

File: exotica_python/src/Exotica.cpp [hunk 11]
```diff
@@ -631,468 +599,450 @@ PYBIND11_MODULE(_pyexotica, module)
...
+    time_indexed_sampling_problem.def("update", &TimeIndexedSamplingProblem::Update);
+    time_indexed_sampling_problem.def("get_space_dim", &TimeIndexedSamplingProblem::getSpaceDim);
+    time_indexed_sampling_problem.def("get_bounds", &TimeIndexedSamplingProblem::getBounds);
+    time_indexed_sampling_problem.def_property("goal_state", &TimeIndexedSamplingProblem::getGoalState, &TimeIndexedSamplingProblem::setGoalState);
+    time_indexed_sampling_problem.def_property("goal_time", &TimeIndexedSamplingProblem::getGoalTime, &TimeIndexedSamplingProblem::setGoalTime);
+    time_indexed_sampling_problem.def_readonly("N", &TimeIndexedSamplingProblem::N);
+    time_indexed_sampling_problem.def_readonly("num_tasks", &TimeIndexedSamplingProblem::NumTasks);
+    time_indexed_sampling_problem.def_readonly("Phi", &TimeIndexedSamplingProblem::Phi);
+    time_indexed_sampling_problem.def_readonly("Inequality", &TimeIndexedSamplingProblem::Inequality);
+    time_indexed_sampling_problem.def_readonly("Equality", &TimeIndexedSamplingProblem::Equality);
```
Comment (VladimirIvan) [inline exotica_python/src/Exotica.cpp]: Lower case (keep Phi)

File: exotica_python/src/Exotica.cpp (hunk 2 above)
Comment (wxmerkt) [reply exotica_python/src/Exotica.cpp]: No, one is an attribute and one is a getter (method). So I will change capitalisation but we cannot make API equal.

**Pull request merged**